src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProjectProvider.cs

```diff
@@ -75,7 +75,7 @@ public NetCorePackageReferenceProjectProvider(IProjectSystemCache projectSystemC
             }
 
             // Check if the project is not CPS capable or if it is CPS capable then it does not have TargetFramework(s), if so then return false
-            if (!hierarchy.IsCapabilityMatch("CPS"))
+            if (!await hierarchy.IsCPSCapabilityComplaintAsync())
```

Comment:
Likely to clash with https://github.com/NuGet/NuGet.Client/pull/3644. 
cc @davkean